### ISSUE TITLE
Internal error (using l2s)

### ISSUE THREAD

**Issue by grafael** [JohnLangford/vowpal_wabbit#1276]
I was trying to reproduce the example described in the document: [Joint Prediction](https://github.com/JohnLangford/vowpal_wabbit/wiki/l2s.pdf) 
That says:
```shell
vw -b 24 -d wsj.train.vw -c --search_task sequence  \
--search 45 --search_neighbor_features -1:w,1:w \
--affix -1w,_1w -f wsj.weights
```

I just changed the `wsj.train` to be my own file. My file contains the following data:
```
1|This
2|is
1|a
3|sequence

1|Another
2|sequence
```

Also, i changed the param `--search` to be equal **3** _(number of labels)_
After running the command I got the error:
```c
vw (search.cc:473): internal error (bug): expecting top namespace to be '131' but it was 128l
```

**Is there anything I'm missing? Or indeed it is a bug?**

**Comment by JohnLangford** [comment]
@hal3 @KaiWeiChang any ideas?

**Comment by JohnLangford** [comment]
This looks like a bug.  I can reproduce via:
echo "1| This" | vw -b 24 -d data -c --search_task sequence --search 2 --search_neighbor_features 1:w

**Comment by JohnLangford** [comment]
Or even just: 
echo "1| f" | vw data --search_task sequence --search 2 --search_neighbor_features 1:w

**Comment by usptact** [comment]
I use L2S to train a NER model. Is there a documentation for all L2S related options? In particular, here are the options I use:

```
vw --data $DATA \
     --learning_rate 0.05 \
     --cache --kill_cache \
     --threads \
     --passes 10 \
     --search_task sequence \
     --search $NUM_LABELS \
     --search_rollin=policy \
     --search_rollout=none \
     --named_labels $LABELS \
     -b 28 \
     --l1=1e-7 \
     -f $MODEL
```

Is this a reasonable setup? The two options that are not very clear to me are the "--search_rollin" and "--search_rollout".

Thanks

**Comment by usptact** [comment]
I created a PR to remove the cout statements that print these lines #1292

**Comment by hal3** [comment]
The bug seems to be: if you ask for neighbor features on a namespace that doesn't exist, bad things happen. In keeping with general VW style, I think the preferred behavior is for this to essentially be ignored. I'll fix this way shortly.

**Comment by hal3** [comment]
I have a fix. One minor comment is that your data example above won't work anyway. You need a space after the "|" in, eg:

`1|This`

**Comment by JohnLangford** [comment]
Should be fixed now.
